### ISSUE TITLE
Add support for tickets

### ISSUE THREAD

**Issue by qeef** [thomersch/openstreetmap-calendar#11]
Currently, for organizing Missing Maps mapathons, we use [Eventbrite](https://www.eventbrite.com/). It would be awesome to switch to osmcal.org. However, we need the ticketing system that enables attend the event with particular category (i.e. beginner mapper, JOSM mapper, validator, etc.).

Is this functionality planned?

**Comment by qeef** [comment]
I did a similar event system we use for small actions, see https://gitlab.com/mapathon-cz/osmevents.

**Comment by thomersch** [comment]
Yes, I've been already thinking about it: I had this idea that event creators can define questions for attendees, which they have to answer to sign up.

So, as the event organiser you could create the event and specify that every attendee needs to answer the question "What kind of participant are you?" and give "Beginner Mapper", "JOSM Mapper" etc answer options.

Would this help you?

**Comment by qeef** [comment]
This would be awesome. One thing yet comes to my mind -- what about capacity? I.e. when there is space for 30 beginner mappers and 10 JOSM mappers?

**Comment by thomersch** [comment]
We could let the users specify quotas for answers. I'll have a look how an implementation may look like.

**Comment by thomersch** [comment]
Update: Work is in progress, I managed to build the question setup form for organisers and prepared already the groundwork for the participants' questionnaire.

There is still some UI work to do, as it is still very, very ugly at the moment, but we're getting there. I hope to be done until end of the week.

**Comment by thomersch** [comment]
Good news: We now have a beta version running on osmcal.org and we already used it to set up our next Hack Weekend.

Creation Form:

<img width="766" alt="Screenshot 2020-02-23 at 19 22 35" src="https://user-images.githubusercontent.com/948605/75117464-ddc21a00-5671-11ea-9e50-3954a994936a.png">

Participants Overview:

<img width="1060" alt="Screenshot 2020-02-23 at 19 21 54" src="https://user-images.githubusercontent.com/948605/75117456-c97e1d00-5671-11ea-8ef8-f3b5e9926e4c.png">

There are a few things still missing:

* There are no quotas.
* You can't edit your own answers, you have to un-join and join again.
* Questions cannot be changed (whether this is desirable is questionable anyway) and choices cannot be added (this is surely useful). Questions also can't be deleted (probably also useful).

**Comment by qeef** [comment]
I really like it, thanks :)

One question -- is it possible to group by a category the list of participants? This will be really appreciated in our use-case.

**Comment by thomersch** [comment]
>  is it possible to group by a category the list of participants?

Oh, this seems useful. How about being able to sort columns? This shouldn't be too much work.

**Comment by qeef** [comment]
I have some feedback here:
1. When I added *text* answer, it took only the first word of the first answer.
2. It would be nice to have 0 survey questions as default.

**Comment by thomersch** [comment]
> 1.  I added text answer, it took only the first word of the first answer.

Ok, that’s a bug. Gonna look into it.

> 2. It would be nice to have 0 survey questions as default.

A survey without any questions doesn’t seem too useful to me...? Why do you want this?

**Comment by thomersch** [comment]
> 1. I added text answer, it took only the first word of the first answer.

I somehow cannot reproduce this...?

<img width="474" alt="Screenshot 2020-06-16 at 11 17 33" src="https://user-images.githubusercontent.com/948605/84756473-2015bb00-afc3-11ea-9602-369f1ad2ae24.png">

**Comment by qeef** [comment]
> > 1. I added text answer, it took only the first word of the first answer.
> 
> Ok, that’s a bug. Gonna look into it.
>
> I somehow cannot reproduce this...?

Hmm, I will look into it. The truth is that I did it just once, so I wil test in a little bit more.

> > 1. It would be nice to have 0 survey questions as default.
> 
> A survey without any questions doesn’t seem too useful to me...? Why do you want this?

When I create an event, it always put 4 questions. What if I would like to create an event without a survey? (Or maybe it's again my fault, like maybe something cached or so?)

**Comment by thomersch** [comment]
> When I create an event, it always put 4 questions. 

This shouldn't be that way :) Normally it should only creates one.

![2020-06-16 13-42-59 2020-06-16 13_44_03](https://user-images.githubusercontent.com/948605/84770233-77258b00-afd7-11ea-969d-1398232e3e10.gif)

Can you tell me what browser and OS you're on?

> maybe something cached or so?

This could be. Maybe your browser still had something very old in-cache. But then again, the form JavaScript is inlined...

**Comment by qeef** [comment]
I use Firefox `68.9.0esr (64-bit)` on `Debian GNU/Linux 10 (buster)`.

**Comment by qeef** [comment]
> > > 1. I added text answer, it took only the first word of the first answer.
> > 
> > 
> > Ok, that’s a bug. Gonna look into it.
> > I somehow cannot reproduce this...?
> 
> Hmm, I will look into it. The truth is that I did it just once, so I wil test in a little bit more.

So I checked it and the problem is when there is `,` in the choice.


>> When I create an event, it always put 4 questions.
>
> This shouldn't be that way :) Normally it should only creates one.

I was not able to reproduce this one again, so that was something on my side.


The last thing is not so big problem -- when I forgot to specify *Start* date, the survey resets. However, when I forgot to specify *Kind*, it does not influence the survey.

**Comment by danieldegroot2** [comment]
@thomersch For Answer Type "Choice", allow users to delete any choices (but one) individually.
